jobs/scheduled/monitor_event_dates.rb

```diff
@@ -46,9 +46,14 @@ def due_reminders(event_date)
       return [] if event_date.event.reminders.blank?
       event_date.event.reminders.split(",").map do |reminder|
         value, unit = reminder.split('.')
-        date = event_date.starts_at - value.to_i.send(unit)
-        { description: reminder, date: date }
-      end.select { |reminder| reminder[:date] <= Time.current }.sort_by { |reminder| reminder[:date] }.drop(event_date.reminder_counter)
+
+        begin
+          date = event_date.starts_at - value.to_i.public_send(unit)
```

Comment:
I do not think there are so many acceptable values for unit (second, minute, hour, day, month and maybe year? plus their plural forms), so what I would do instead is check for those, instead of the begin-rescue dance.

This gives better control over what gets called. In the future, when this will no longer be an experimental feature I guess we will need the units list for validation anyway.

Reply:
Oh yes agreed. Now that we have the `validate_reminder_unit` check, the begin/rescue can probably be removed.

> plus their plural forms

Looks like `ActiveSupport::Duration::PARTS` includes only the plural forms. So I guess we need to allow the singular form as well?

```
[1] pry(main)> ActiveSupport::Duration::PARTS
=> [:years, :months, :weeks, :days, :hours, :minutes, :seconds]
```

Reply:
No the builder will output `1.minutes` and I think it's fine, at least for now. Maybe we can deal with this later, but any error coming from this is manual input error.